### ISSUE TITLE
Add buttons to start and stop all adapters

### ISSUE THREAD

**Issue by tenthe** [apache/streampipes#1113]
### Body

# Description
A user should have the option to start / stop all adapters together.

You can have a look at the feature in the  pipeline overview (`start all pipelines` / `stop all pipelines`).
The same functionality should be available for the adapters in the connect view.

# Mentoring
As this ticket is marked as good first issue: one of @dominikriemer or @tenthe are happy to provide help for getting started, just tag (one of) them if you want to start working on this issue and need some help.
If you are familiar with Angular, this is a good issue to get started with UI development in StreamPipes. 

### StreamPipes Committer

I acknowledge that I am a maintainer/committer of the Apache StreamPipes project.

**Comment by purplesmurf45** [comment]
Hi @bossenti, I’d like to work on this issue. Could you assign this task to me? Any resources related to the issue would help a lot!

**Comment by bossenti** [comment]
Sure 🙂 

The pipeline overview that is mentioned in the description as reference can be found here: https://github.com/apache/streampipes/blob/dev/ui/src/app/pipelines/pipelines.component.html & https://github.com/apache/streampipes/blob/dev/ui/src/app/pipelines/pipelines.component.ts

The corresponding adapter view where this functionality should be added is this one: https://github.com/apache/streampipes/tree/dev/ui/src/app/connect/components/existing-adapters

Here is a documentation on how to develop the UI: https://cwiki.apache.org/confluence/display/STREAMPIPES/UI

Do you need anything more?

**Comment by purplesmurf45** [comment]
This was helpful! I have started working on it. I will update you if I face any issues 🙂

**Comment by purplesmurf45** [comment]
@bossenti @dominikriemer 
I am trying to access the running flag of adapters but I am facing this error - 
```
Property 'running' does not exist on type 'AdapterDescriptionUnion'.
  Property 'running' does not exist on type 'GenericAdapterSetDescription'.ts(2339)
```

On further inspection, I noticed that this value is being accessed by a different function and that function is working fine. When I printed out the attributes of AdapterDescriptionUnion, I could see that the running attribute does exist and has a value associated with it. Can someone help me understand why this is happening? And is there any workaround for getting the running state of adapters?

**Comment by bossenti** [comment]
Hm, I'm not really an expert with respect to our UI
Maybe @tenthe or @smlabt can help you here

**Comment by purplesmurf45** [comment]
@tenthe @smlabt could you help me with this?

**Comment by dominikriemer** [comment]
@purplesmurf45 thanks for the good explanation!
@tenthe 
I had a look at the adapter data models and it seems the `running` attribute is only avaiable for `AdapterStreamDescription` classes. Is there a reason why we do not have the `running` field at the top-level class `AdapterDescription`?

**Comment by tenthe** [comment]
Hello @dominikriemer, hello @purplesmurf45,
the reason was that set adapters were started as soon as the pipeline was invoked. So users could not manually start or stop a set adapter.
@purplesmurf45 for this issue you can ignore the `...SetDescription` as we will remove set adapters soon.
Hope this helps. Please let us know if there are any other issues.

**Comment by purplesmurf45** [comment]
Hello @tenthe, I went over #1290 and #1289... I understand the need to remove set adapters. But even when I'm trying to start/stop specific `MachineDataSimulator` adaptors which are stream adapters, I am getting the same 
```
Property 'running' does not exist on type 'AdapterDescriptionUnion'.
  Property 'running' does not exist on type 'GenericAdapterSetDescription'.ts(2339) 
```
error. Any idea why this is happening? For now, I'm passing the list of all adapters to the `start all adapters` / `stop all adapters` functions and it is working fine. If I was able to access the `running` field, I could only show the adapters to be started or stopped instead of displaying the entire list of adapters (present in My Adapters) in the dialog box...
